frontend/src/components/pages/GlobalForms/WaiverAddSectionModal.tsx

```diff
@@ -0,0 +1,100 @@
+import {
+  Box,
+  Button,
+  Checkbox,
+  HStack,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Text,
+  Textarea,
+} from "@chakra-ui/react";
+import React from "react";
+import { WaiverClause } from "../../../types/AdminTypes";
+
+type WaiverAddSectionModalProps = {
+  isOpen: boolean;
+  onClose: () => void;
+  onAddWaiverSectionClick: (newClause: WaiverClause) => void;
+};
+
+const WaiverAddSectionModal = ({
+  isOpen,
+  onClose,
+  onAddWaiverSectionClick,
+}: WaiverAddSectionModalProps): JSX.Element => {
+  const [newClauseText, setNewClauseText] = React.useState("");
+  const [clauseRequired, setClauseRequired] = React.useState(false);
+
+  const handleInputChange = (e: any) => {
+    const inputValue = e.target.value;
+    setNewClauseText(inputValue);
+  };
+
+  const handleCheckboxChange = () => {
+    setClauseRequired(!clauseRequired);
```

Comment:
Since you're relying on the old state, I think it's best practise to do it like this:
`setClauseRequired((prevState) => !prevState);`